sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -361,6 +362,41 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
   }
 
   /**
+   * Loads an `Dataset[String]` storing CSV rows and returns the result as a `DataFrame`.
+   *
+   * Unless the schema is specified using `schema` function, this function goes through the
+   * input once to determine the input schema.
+   *
+   * @param csvDataset input Dataset with one CSV row per record
+   * @since 2.2.0
+   */
+  def csv(csvDataset: Dataset[String]): DataFrame = {
```

Comment:
Sure. Actually, there is a JIRA and closed PR https://github.com/apache/spark/pull/13460 and [SPARK-15615](https://issues.apache.org/jira/browse/SPARK-15615) where I was negative because it can be easily worked around.

However, I am fine if we are promoting to use datasets instead of RDDs for some advantages such as [SPARK-18362](https://issues.apache.org/jira/browse/SPARK-18362).

cc @pjfanning, could you reopen and proceed your PR if we are all fine?